nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/main/java/org/apache/nifi/processors/standard/CaptureChangeMySQL.java

```diff
@@ -0,0 +1,928 @@
+        pds.add(HOSTS);
+        pds.add(DRIVER_NAME);
+        pds.add(DRIVER_LOCATION);
+        pds.add(USERNAME);
+        pds.add(PASSWORD);
+        pds.add(DATABASE_NAME_PATTERN);
+        pds.add(TABLE_NAME_PATTERN);
+        pds.add(CONNECT_TIMEOUT);
+        pds.add(DIST_CACHE_CLIENT);
+        pds.add(RETRIEVE_ALL_RECORDS);
+        pds.add(INIT_SEQUENCE_ID);
+        pds.add(INIT_BINLOG_FILENAME);
+        pds.add(INIT_BINLOG_POSITION);
+        propDescriptors = Collections.unmodifiableList(pds);
+    }
+
+
+    @Override
+    public Set<Relationship> getRelationships() {
+        return relationships;
+    }
+
+    @Override
+    protected List<PropertyDescriptor> getSupportedPropertyDescriptors() {
+        return propDescriptors;
+    }
+
+    @Override
+    protected PropertyDescriptor getSupportedDynamicPropertyDescriptor(final String propertyDescriptorName) {
+        if (INIT_BINLOG_FILENAME.getName().equals(propertyDescriptorName)) {
+            return INIT_BINLOG_FILENAME;
+        }
+        if (INIT_BINLOG_POSITION.getName().equals(propertyDescriptorName)) {
+            return INIT_BINLOG_POSITION;
+        }
+        return null;
+    }
+
+    @OnScheduled
+    public void setup(ProcessContext context) {
+
+        final ComponentLog logger = getLogger();
+
+        final StateManager stateManager = context.getStateManager();
+        final StateMap stateMap;
+
+        try {
+            stateMap = stateManager.getState(Scope.LOCAL);
+        } catch (final IOException ioe) {
+            logger.error("Failed to retrieve observed maximum values from the State Manager. Will not attempt "
+                    + "connection until this is accomplished.", ioe);
+            context.yield();
+            return;
+        }
+
+        Map<String, String> statePropertiesMap = new HashMap<>(stateMap.toMap());
+
+        // Pre-store initial sequence ID if none has been set in the state map
+        final PropertyValue initSequenceId = context.getProperty(INIT_SEQUENCE_ID);
+        statePropertiesMap.computeIfAbsent(EventWriter.SEQUENCE_ID_KEY, k -> initSequenceId.isSet() ? initSequenceId.evaluateAttributeExpressions().getValue() : "0");
+
+
+        boolean getAllRecords = context.getProperty(RETRIEVE_ALL_RECORDS).asBoolean();
+
+        // Set current binlog filename to whatever is in State, falling back to the Retrieve All Records then Initial Binlog Filename, if no State variable is present
+        currentBinlogFile = stateMap.get(BinlogEventInfo.BINLOG_FILENAME_KEY);
+        if (StringUtils.isEmpty(currentBinlogFile)) {
+            if (!getAllRecords && context.getProperty(INIT_BINLOG_FILENAME).isSet()) {
+                currentBinlogFile = context.getProperty(INIT_BINLOG_FILENAME).evaluateAttributeExpressions().getValue();
+            } else {
+                // If we're starting from the beginning of all binlogs, the binlog filename must be the empty string (not null)
+                currentBinlogFile = "";
```

Comment:
Makes sense, but how do we know if we're at "current" or some other arbitrary file/position? Or do you mean make a property available to the user to choose "beginning, current, or some arbitrary location"?